### PR TITLE
update docs / combine local and remote config / update demo data

### PR DESCRIPTION
**Pull request by mwarning**
I have tried to reconcile the usage of the remote  `.versions.json` file with the local `config.js` file.

=> Missing `version` and `default_version` are taken from `.versions.json`.

### CODE REVIEW

**Comment by mwarning** [comment]
@aparcar what do you think? That should be compatible with the OpenWrt setup, allow other parties to use it without too much effiort and enable a demo via github.

**Comment by mwarning** [comment]
I noticed that the OpenWrt setup is not very uniform:
- snapshot version are added in the frontend/javascript code
- the "snapshot" images are placed in a specifc location
- the field in the remote .versions.json are not called the same as in the config.json

The hardcoded paths also make it hard to allow a standalone demonstration mode that does not require Internet access,

So I have opted to create a "mode" setting that can be either "demo", "openwrt" or not be set. It is documented in the config.js and README.md files.

**Comment by aparcar** [comment]
Hi @mwarning , thanks for your continuous effort regarding the firmware selector!

From my point of view we should drop the GitLab CI stuff entirely and use GitHub Action only, part of my motivation here was to have a static page instead of an active service running on our infrastructure. Could you please drop all GitLab stuff and instead use the [GitHub CI](https://github.com/openwrt/firmware-selector-openwrt-org/blob/master/.github/workflows/static.yml)?

The rest looks fine to me, I'll give it a local run before merging.

Regarding the SNAPSHOT stuff, they are always handled awkwardly different in OpenWrt and therefore I'd just do the downstream coding to make things work. Adding SNAPSHOTs upstream causes path issues since all versions but SNAPSHOT are listed under `releases/<version>/`. Secondly there are motivations to hide all `-SNAPHOT` versions since users would consider it stable and then break their devices. 

Please keep the downstream code for that in place.

File: www/index.js [hunk 1]
```diff
@@ -742,43 +757,82 @@ async function init() {
...
+              urls[version] = `${config.image_url}/snapshots/`;
+            } else {
+              urls[version] = `${config.image_url}/releases/${version}`;
+            }
+          }
+        }
 
-  console.log(versions);
+        return {
+          versions: versions,
```
Comment (aparcar) [inline www/index.js:801]: Shouldn't this be the following?
```javascript 
          versions: obj.versions_list,
```

File: www/index.js [hunk 2]
```diff
@@ -742,43 +757,82 @@ async function init() {
...
 
-  let versions = await fetch(config.image_url + "/.versions.json", {
-    cache: "no-cache",
-  })
-    .then((obj) => {
-      return obj.json();
+  config.overview_urls = {};
+  config.image_urls = {};
+
+  if (config.mode == "demo") {
```
Comment (aparcar) [inline www/index.js:763]: I think this demo mode is a bit over engineered. Can't we just point it to a path which is either a remote URL or a local file? In both cases it'll look for `.versions.json` anyway.

Comment (mwarning) [reply www/index.js:763]: Well, it is ugly indeed.

What we could do is make the `.versions.json` mandatory for all setups  (thus removing `versions` and `default_version` from `config.json`). Then we could add an optional image_url in `.versions.json`. That would remove the need for any `mode`.

**Comment by mwarning** [comment]
@aparcar I have added a commit "rework":
- removed "mode" setting
- add optional image_url_override to `.versions.json` to allow the demo to work without Internet access.

The `versions` and `default_version` configuration options are now optional.

**Comment by mwarning** [comment]
> Hi @mwarning , thanks for your continuous effort regarding the firmware selector!
> 
> From my point of view we should drop the GitLab CI stuff entirely and use GitHub Action only, part of my motivation here was to have a static page instead of an active service running on our infrastructure. Could you please drop all GitLab stuff and instead use the [GitHub CI](https://github.com/openwrt/firmware-selector-openwrt-org/blob/master/.github/workflows/static.yml)?
> 
I try to avoid working on pipelines as much as possible since it is no fun for me. Can you take that? I can certainly remove any modification of the CI stuff from this MR.

**Comment by aparcar** [comment]
Great, works for me now, please drop the CI stuff and I'll merge it :)

**Comment by mwarning** [comment]
> Great, works for me now, please drop the CI stuff and I'll merge it :)

Done. We can refine in future MRs.

**Comment by mwarning** [comment]
hm, the html5validator does not know the (well supported) CSS attribute called "color-scheme".

**Comment by aparcar** [comment]
> hm, the html5validator does not know the (well supported) CSS attribute called "color-scheme".

I commented on an existing issue here https://github.com/svenkreiss/html5validator/issues/124#issuecomment-2340389774

**Pull request merged**